cni/pkg/util/util.go

```diff
@@ -19,21 +19,30 @@ import (
 	"encoding/json"
 	"fmt"
 	"os"
+	"path/filepath"
+	"strings"
 
 	"github.com/fsnotify/fsnotify"
 
 	"istio.io/istio/pkg/file"
 )
 
+type watchPrefix struct {
+	watcher *fsnotify.Watcher
+	prefix  string
+}
+
 // Creates a file watcher that watches for any changes to the directory
-func CreateFileWatcher(dirs ...string) (watcher *fsnotify.Watcher, fileModified chan bool, errChan chan error, err error) {
```

Comment:
why replace `...string` with `[]string`?

Reply:
because if we have two arguments, first of type `string` and second of type `...string`, go interprets it as a single one as of type `...string`, and thus accepts only one argument as a result. https://go.dev/play/p/HuASlAWYUl2